pydantic/functional_serializers.py

```diff
@@ -177,55 +177,52 @@ def __get_pydantic_core_schema__(self, source_type: Any, handler: GetCoreSchemaH
 
 
 if TYPE_CHECKING:
-    _PartialClsOrStaticMethod: TypeAlias = Union[classmethod[Any, Any, Any], staticmethod[Any, Any], partialmethod[Any]]
-    _PlainSerializationFunction = Union[_core_schema.SerializerFunction, _PartialClsOrStaticMethod]
-    _WrapSerializationFunction = Union[_core_schema.WrapSerializerFunction, _PartialClsOrStaticMethod]
-    _PlainSerializeMethodType = TypeVar('_PlainSerializeMethodType', bound=_PlainSerializationFunction)
-    _WrapSerializeMethodType = TypeVar('_WrapSerializeMethodType', bound=_WrapSerializationFunction)
+    _Partial: TypeAlias = 'partial[Any] | partialmethod[Any]'
 
+    AnyFieldPlainSerializer: TypeAlias = 'core_schema.SerializerFunction | _Partial'
+    """A field serializer method or function in `plain` mode."""
 
-@overload
-def field_serializer(
-    field: str,
-    /,
-    *fields: str,
-    return_type: Any = ...,
-    when_used: Literal['always', 'unless-none', 'json', 'json-unless-none'] = ...,
-    check_fields: bool | None = ...,
-) -> Callable[[_PlainSerializeMethodType], _PlainSerializeMethodType]: ...
```

Comment:
Why did we get rid of this `overload`?

Reply:
This overload was used to account for the following usage:

```python
from pydantic import BaseModel, field_serializer


class Model(BaseModel):
    a: int

    @field_serializer('a')
    ...
```

that is, when `mode` wasn't specified (and would default to `'plain'`). But this isn't necessary with the updated overload I added:

```python
@overload
def field_serializer(
    field: str,
    /,
    *fields: str,
    mode: Literal['plain'] = ...,
    return_type: Any = ...,
    when_used: WhenUsed = ...,
    check_fields: bool | None = ...,
) -> Callable[[_AnyFieldPlainSerializerT], _AnyFieldPlainSerializerT]: ...
```

As type checkers will use this overload when using both `@field_serializer('a')` _and_ `@field_serializer('a', mode='plain')` -- thanks to `mode` being specified as "optional" (with a `...` default value).

Reply:
Awesome! Good to know 🤓